File: LLM-Translations/GPT-4/avatar/Java/C/atcoder_AGC002_B.c
#include <stdio.h>
#include <stdbool.h>

typedef struct {
    int num;
    bool red;
} Box;

void moveTo(Box *src, Box *dest) {
    if (src->num == 0) {
        return;
    } else if (src->num == 1) {
        src->num = 0;
        dest->num++;
        if (src->red) {
            dest->red = true;
        }
        src->red = false;
    } else {
        src->num--;
        dest->num++;
        if (src->red) {
            dest->red = true;
        }
    }
}

int main() {
    int N, M;
    scanf("%d%d", &N, &M);
    Box B[N];
    B[0] = (Box){1, true};
    for (int i = 1; i < N; i++) {
        B[i] = (Box){1, false};
    }

    for (int i = 0; i < M; i++) {
        int x, y;
        scanf("%d%d", &x, &y);
        x--; y--;
        moveTo(&B[x], &B[y]);
    }

    int counter = 0;
    for (int i = 0; i < N; i++) {
        if (B[i].red) {
            counter++;
        }
    }

    printf("%d\n", counter);
    return 0;
}

// //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/Python/atcoder_AGC002_B.py

class Box:
    def __init__(self, n, r):
        self.num = n
        self.red = r

    def move_to(self, other):
        if self.num == 0:
            return
        elif self.num == 1:
            self.num = 0
            other.num += 1
            if self.red:
                other.red = True
            self.red = False
        else:
            self.num -= 1
            other.num += 1
            if self.red:
                other.red = True


def main():
    N, M = map(int, input().split())
    B = [Box(1, False) for _ in range(N)]
    B[0] = Box(1, True)

    for _ in range(M):
        x, y = map(int, input().split())
        x -= 1
        y -= 1
        B[x].move_to(B[y])

    counter = 0
    for b in B:
        if b.red:
            counter += 1

    print(counter)


if __name__ == '__main__':
    main()


##End of Code
